EIPS/eip-4671.md

```diff
@@ -0,0 +1,256 @@
+<!-- The below code snippet is automatically added from ./contracts/IERC4671Consensus.sol -->
+```solidity
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./IERC4671.sol";
+
+interface IERC4671Consensus is IERC4671 {
+    /// @notice Get voters addresses for this consensus contract
+    /// @return Addresses of the voters
+    function voters() external view returns (address[] memory);
+
+    /// @notice Cast a vote to mint a token for a specific address
+    /// @param owner Address for whom to mint the token
+    function approveMint(address owner) external;
+
+    /// @notice Cast a vote to invalidate a specific token
+    /// @param tokenId Identifier of the token to invalidate
+    function approveInvalidate(uint256 tokenId) external;
+}
+```
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+### NTT Store
+
+NTTs are meant to be fetched by third-parties, which is why there need to be a convienent way for users to expose some or all of their NTTs. We achieve this result using a store which must implement the following interface.
+
+<!-- AUTO-GENERATED-CONTENT:START (CODE:syntax=solidity&src=./contracts/IERC4671Store.sol) -->
+<!-- The below code snippet is automatically added from ./contracts/IERC4671Store.sol -->
+```solidity
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./IERC165.sol";
+
+interface IERC4671Store is IERC165 {
+    // Event emitted when a NTT contract is added to the owner's records
+    event Added(address owner, address ntt);
+
+    // Event emitted when a NTT contract is removed from the owner's records
+    event Removed(address owner, address ntt);
+
+    /// @notice Add a NTT contract address to the caller's record
+    /// @param ntt Address of the NTT contract to add
+    function add(address ntt) external;
+
+    /// @notice Remove a NTT contract from the caller's record
+    /// @param ntt Address of the NTT contract to remove
+    function remove(address ntt) external;
+
+    /// @notice Get all the NTT contracts for a given owner
+    /// @param owner Address for which to retrieve the NTT contracts
+    function get(address owner) external view returns (address[] memory);
+}
+```
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+## Rationale
+
+### On-chain vs Off-chain
+
+A decision was made to keep the data off-chain (via `tokenURI()`) for two main reasons: 
+* Non-Tradable Tokens represent personal possessions. Therefore, there might be cases where the data should be encrypted. The standard should not outline decisions about encryption because there are just so many ways this could be done, and every possibility is specific to the use-case.
+* Non-Tradable Tokens must stay generic. There could have been a possibility to make a `MetadataStore` holding the data of NTTs in an elegant way, unfortunately we would have needed a support for generics in solidity (or struct inheritance), which is not available today.
+
+## Implementation
```

Comment:
```suggestion
## Reference Implementation
```

Reply:
Thank you for taking the time to review!